### PR TITLE
#3524 Convert SpinnerLanguagesAdapter to kotlin - converted to kotlin

### PR DESCRIPTION
**Pull request by macgills**
**Description (required)**

Fixes ##3524 Convert SpinnerLanguagesAdapter to kotlin

What changes did you make and why?

Converted to kotlin for easier merging to structured data

**Tests performed (required)**

Tested betaDebug on Nexus5X with API level 8.1.0.

### CODE REVIEW

**Comment by maskaravivek** [comment]
As this PR uses Kotlin Android Extensions, let us wait for #3530 to be finalized before merging this.

File: app/src/main/java/fr/free/nrw/commons/upload/SpinnerLanguagesAdapter.kt [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+                !(selectedLanguages.containsKey(languageCode) && languageCode != selectedLangCode)
+            if (languageCode.isEmpty()) {
+                tv_language.text = StringUtils.capitalize(languageName)
+                tv_language.textAlignment = View.TEXT_ALIGNMENT_CENTER
+            } else {
+                tv_language.text =
+                    "${StringUtils.capitalize(languageName)}" +
+                            " [${LangCodeUtils.fixLanguageCode(languageCode)}]"
+            }
+
```
Comment (misaochan) [inline app/src/main/java/fr/free/nrw/commons/upload/SpinnerLanguagesAdapter.kt]: Unnecessary whitespace?

File: app/src/main/java/fr/free/nrw/commons/upload/SpinnerLanguagesAdapter.kt [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+import androidx.annotation.LayoutRes
+import androidx.core.os.ConfigurationCompat
+import fr.free.nrw.commons.R
+import fr.free.nrw.commons.utils.BiMap
+import fr.free.nrw.commons.utils.LangCodeUtils
+import kotlinx.android.extensions.LayoutContainer
+import kotlinx.android.synthetic.main.row_item_languages_spinner.*
+import org.apache.commons.lang3.StringUtils
+import java.util.*
+
```
Comment (misaochan) [inline app/src/main/java/fr/free/nrw/commons/upload/SpinnerLanguagesAdapter.kt]: Could we get Javadocs for new classes please? They should definitely have been in the original to begin with, but we have to start somewhere - if we are rewriting an entire class I think it is a good time to start including Javadocs as well.

File: app/src/main/java/fr/free/nrw/commons/upload/SpinnerLanguagesAdapter.kt [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+    }
+
+    fun getIndexOfUserDefaultLocale(context: Context): Int {
+        return languageCodesList.indexOf(context.locale.language)
+    }
+
+    fun getIndexOfLanguageCode(languageCode: String): Int {
+        return languageCodesList.indexOf(languageCode)
+    }
+
```
Comment (misaochan) [inline app/src/main/java/fr/free/nrw/commons/upload/SpinnerLanguagesAdapter.kt]: Unnecessary whitespace?

File: app/src/main/java/fr/free/nrw/commons/upload/SpinnerLanguagesAdapter.kt [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+    context: Context,
+    private val selectedLanguages: BiMap<*, String>
+) : ArrayAdapter<Any?>(context, -1) {
+
+    private val languageNamesList: List<String>
+    private val languageCodesList: List<String>
+
+    init {
+        val sortedLanguages =
+            Locale.getAvailableLocales().map(::Language).sortedBy { it.locale.displayName }
```
Comment (misaochan) [inline app/src/main/java/fr/free/nrw/commons/upload/SpinnerLanguagesAdapter.kt]: Does our codestyle mandate a specific line length? I find that this sort of formatting tends to hinder readability instead of helping it, but if the codestyle itself specifies this then I could be wrong.

Comment (macgills) [reply app/src/main/java/fr/free/nrw/commons/upload/SpinnerLanguagesAdapter.kt]: Codestyle specifies it, [link](https://github.com/google/styleguide/blob/305ac8725a166ed42e3f5dd3f80d6de2cf840ef1/intellij-java-google-style.xml#L28)

File: app/src/main/java/fr/free/nrw/commons/upload/SpinnerLanguagesAdapter.kt (hunk 2 above)
Comment (macgills) [reply app/src/main/java/fr/free/nrw/commons/upload/SpinnerLanguagesAdapter.kt]: Is this satisfactory? I would never have written javadocs for any sustained length of time as every company I have worked in loathed them

Comment (misaochan) [reply app/src/main/java/fr/free/nrw/commons/upload/SpinnerLanguagesAdapter.kt]: Yes, thank you! :)
